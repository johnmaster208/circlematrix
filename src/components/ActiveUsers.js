import React from 'react'
import PropTypes from 'prop-types'


const ActiveUsers = (props) => {
  const { users } = props;
  return(
    <div>
      Active users:
      <ul>
        {users.map((item, index) => {
          return(<li key={index}>{item}</li>)
        })}
      </ul>
    </div>
  );

};

ActiveUsers.propTypes = {
  users: PropTypes.array.isRequired
}

export default ActiveUsers
