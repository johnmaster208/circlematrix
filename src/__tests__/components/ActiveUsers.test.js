import React from 'react';
import ActiveUsers from '../../components/ActiveUsers';
import { shallow } from 'enzyme';

describe('ActiveUsers component', () => {

    const props = {
        users: [],
        currentUser: "aasdfa_qereveq134"
    };

    it('Renders without crashing...', () => {
        const wrapper = shallow(
            <ActiveUsers
                {...props } />
        );
        expect(wrapper).toMatchSnapshot();
    });

});