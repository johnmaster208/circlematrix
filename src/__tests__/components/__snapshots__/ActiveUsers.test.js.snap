// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActiveUsers component Renders without crashing... 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ActiveUsers
    currentUser="aasdfa_qereveq134"
    users={Array []}
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        "Active users:",
        <ul
          className="active-users list-group"
>
          
</ul>,
      ],
    },
    "ref": null,
    "rendered": Array [
      "Active users:",
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [],
          "className": "active-users list-group",
        },
        "ref": null,
        "rendered": Array [],
        "type": "ul",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          "Active users:",
          <ul
            className="active-users list-group"
>
            
</ul>,
        ],
      },
      "ref": null,
      "rendered": Array [
        "Active users:",
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [],
            "className": "active-users list-group",
          },
          "ref": null,
          "rendered": Array [],
          "type": "ul",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
